src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,426 @@
+                    if ((potentiaCitizenNameParts[0].equals(citizenNameParts[0])) && (potentiaCitizenNameParts[1].equals(citizenNameParts[1])))
+                    {
+                        lastNameSet.add(citizenNameParts[2]);
+                    }
+                }
+                return lastNameSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[2])).collect(Collectors.toList());
+           }
+        }
+        return Collections.emptyList();
+    }
+
+    public Object parse(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @Nullable final BlockPos pos,
+            @NotNull final List<ActionMenuHolder> parsedHolders,
+            final String potentialArgumentValue)
+    {
+        // TODO: selector support, such as used by CommandKill to find player
+        // Entity entity = <net.minecraft.command.CommandBase>.getEntity(server, sender, args[0]);
+
+        switch (this)
+        {
+            case INTEGER:
+            case COORDINATE_X:
+            case COORDINATE_Y:
+            case COORDINATE_Z:
+                return Ints.tryParse(potentialArgumentValue);
+            case BOOLEAN:
+                return parseBoolean(potentialArgumentValue);
+            case PLAYER:
+                return parsePlayerValue(server, potentialArgumentValue);
+            case COLONY:
+                return parseColonyValue(sender, potentialArgumentValue);
+            case CITIZEN:
+                return parseCitizenDataValue(parsedHolders, potentialArgumentValue);
+            default:
+                throw new IllegalStateException("Unimplemented ActionArgumentType parsing");
+        }
+    }
+
+    private Boolean parseBoolean(final String potentialArgumentValue)
+    {
+        if ("true".equalsIgnoreCase(potentialArgumentValue))
+        {
+            return Boolean.TRUE;
+        }
+        if ("t".equalsIgnoreCase(potentialArgumentValue))
+        {
+            return Boolean.TRUE;
+        }
+        if ("yes".equalsIgnoreCase(potentialArgumentValue))
+        {
+            return Boolean.TRUE;
+        }
+        if ("y".equalsIgnoreCase(potentialArgumentValue))
+        {
+            return Boolean.TRUE;
+        }
+        if ("1".equalsIgnoreCase(potentialArgumentValue))
+        {
+            return Boolean.TRUE;
+        }
+        if ("false".equalsIgnoreCase(potentialArgumentValue))
+        {
+            return Boolean.FALSE;
+        }
+        if ("f".equalsIgnoreCase(potentialArgumentValue))
+        {
+            return Boolean.FALSE;
+        }
+        if ("no".equalsIgnoreCase(potentialArgumentValue))
+        {
+            return Boolean.FALSE;
+        }
+        if ("n".equalsIgnoreCase(potentialArgumentValue))
+        {
+            return Boolean.FALSE;
+        }
+        if ("0".equalsIgnoreCase(potentialArgumentValue))
+        {
+            return Boolean.FALSE;
+        }
+        return null;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Null is returned but a "Boolean" is expected. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2447)